### PR TITLE
Add documentation about CI/CD

### LINKED ISSUE #21: Documentation:  Add README.md to workflows folder
### Please select a category the issue is focused on?

Other

### Let us know where something needs a refresh or put your idea here!

`README.md` should be added to the `workflows` folder. For each workflow in the `workflows` folder it should provide:
- A short description of the workflow.
- Prerequisites for using the workflow.

### PR DESCRIPTION
**Pull request by cicdguy**
Add a README for the CI/CD workflows.

Fixes https://github.com/pharmaverse/admiraltemplate/issues/21

### CODE REVIEW

File: .github/workflows/README.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+Most workflows have a `BEGIN boilderplate steps` and `END boilderplate steps` section within them which define some standard steps required for installing system dependencies, R and R packages which serve as dependencies for the package.
+
+The underlying mechanisms for installing R and Pandoc are defined in [`r-lib/actions`][r-lib-actions], while the installation of system dependencies and R package dependencies is managed via the [Staged Dependencies GitHub Action][sd-action]. The latter is used in conjunction with the [`staged_dependencies.yaml`](../../staged_dependencies.yaml) file in order to install dependencies that are in the _same stage of development_ as the current package. You can read more about how it works [here](sd-repo). Note that the latter is not necesary for this workflow to work and is completely optional.
+
+Following the installation of system dependencies, R, and package dependencies, each workflow does something different.
+
+### [`check-templates.yml`](check-templates.yml)
+
+This workflow checks for templates.
```
Comment (bms63) [inline .github/workflows/README.md]:
```suggestion
This workflow checks for issues within template scripts.  For example, in admiral package there are several template scripts with admiral-based functions showing how to build certain ADaM datasets.   As we update the admiral functions, we want to make sure these template scripts execute appropriately.  Functions in the template scripts that are deprecated or used inappropriately will cause this workflow to fail. 
```

File: .github/workflows/README.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+This workflow measures code coverage for unit tests and reports the code coverage as a percentage of the _total number of lines covered by unit tests_ vs. the _total number of lines in the codebase_.
+
+The [`covr`][covr] R package is used to calculate the coverage.
+
+Report summaries and badges for coverage are generated using a series of other GitHub Actions.
+
+### [`links.yml`](links.yml)
+
+This workflow checks whether URLs embedded in code and documentation are valid. Invalid URLs reults in workflow failures. This workflow uses [`lychee`][lychee] to detect broken links.
```
Comment (bms63) [inline .github/workflows/README.md]:
```suggestion
This workflow checks whether URLs embedded in code and documentation are valid. Invalid URLs reults in workflow failures. This workflow uses [`lychee`][lychee] to detect broken links.   Occasionally this check will detect false positives of urls that look like urls.  To remedy, please add this false positive to the `.lycheeignore` file.   
```

File: .github/workflows/README.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+When a new release of the package is made, this workflow executes to create a validation report via [theValidatoR][validation]. The PDF report is then attached to the release within GitHub.
+
+### [`readme-render.yml`](readme-render.yml)
+
+If your codebase uses a `README.Rmd` file], then this workflow will automatically render a `README.md` and commit it to your branch.
+
+### [`spellcheck.yml`](spellcheck.yml)
+
+Spellchecks are performed by this workflow, and the [`spelling`][spelling] R package is used to detect spelling mistakes. Failed workflows typically indicate misspelled words.
```
Comment (bms63) [inline .github/workflows/README.md]:
```suggestion
Spellchecks are performed by this workflow, and the [`spelling`][spelling] R package is used to detect spelling mistakes. Failed workflows typically indicate misspelled words.  In the `inst/WORDLIST` file, you can add words and or acronyms that you want the spell check to ignore, for example occds is not an English word but a common acronym used within Pharma.  The workflow will flag this until a user adds it to the `inst/WORDLIST`.
```

**Comment by bms63** [comment]
Well here is the presentation for R/Pharma!  :)  Looks awesome! 

 I'm thinking this might be handy our developer vignettes, almost verbatim, as these worfklows generate lots of questions from new developers.  What do you think @thomas-neitmann @rossfarrugia @bundfussr - a new vignette for the cause!

File: .github/workflows/README.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
+# GitHub Actions Workflows
+
+This directory contains GitHub Actions CI/CD workflows, most of which are also used by `admiral`'s extensions.
```
Comment (bms63) [inline .github/workflows/README.md]:
```suggestion
This directory contains GitHub Actions Continuous integration/continuous delivery (CI/CD) workflows, most of which are also used by `admiral`'s extensions.
```

File: .github/workflows/README.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+Workflows defined here are responsible for assuring high package quality standards without compromising performance, security, or reproducibility.
+
+## What these workflows do?
+
+Most workflows have a `BEGIN boilderplate steps` and `END boilderplate steps` section within them which define some standard steps required for installing system dependencies, R and R packages which serve as dependencies for the package.
```
Comment (bms63) [inline .github/workflows/README.md]:
```suggestion
Most workflows have a `BEGIN boilderplate steps` and `END boilderplate steps` section within them which define some standard steps required for installing system dependencies, R version and R packages which serve as dependencies for the package.
```

**Comment by cicdguy** [comment]
FYI, see reusable workflows in action here: https://github.com/pharmaverse/admiralonco/pull/75

This basically means that workflows can be centrally defined and imported everywhere else by other extensions. This means that one needs to modify workflows only in one specific location (here), and they will be automatically applied to all repos that use these workflows. No need to copy-paste to all repos every time a change is made to a workflow. Keep things DRY.

**Comment by rossfarrugia** [comment]
@bms63 or rather than a new vignette why not add to https://pharmaverse.github.io/admiral/articles/pr_review_guidance.html#review-criteria?

File: .github/workflows/README.md [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+
+This workflow performs `R CND check` for the package. Failed workflows are typically indicative of problems encountered during the check, and therefore an indication that the package does not meet quality standards.
+
+### [`r-pkg-validation.yml`](r-pkg-validation.yml)
+
+When a new release of the package is made, this workflow executes to create a validation report via [theValidatoR][validation]. The PDF report is then attached to the release within GitHub.
+
+### [`readme-render.yml`](readme-render.yml)
+
+If your codebase uses a `README.Rmd` file], then this workflow will automatically render a `README.md` and commit it to your branch.
```
Comment (rossfarrugia) [inline .github/workflows/README.md:52]: very cool @dinakar29 - all reads well and makes sense to me. the only minor thing i spotted is a seemingly stray `]` here

**Comment by bundfussr** [comment]
@dinakar29 , I think the reusable workflows are great. I wonder if we should move the workflows to a separate repository and keep only `common.yml` in admiraltemplate. Otherwise, all the workflows will be copied if a new repository is created based on admiraltemplate.

**Comment by cicdguy** [comment]
Moving to a new repository makes total sense. We need the repository to look like admiraltemplate, however. Similar structure is preferred. It could be called admiralci or something.

File: .github/workflows/README.md (hunk 6 above)
Comment (cicdguy) [reply .github/workflows/README.md:52]: Good find. I'll remove this.

**Comment by bms63** [comment]
Is a reusable workflow like calling the repo's workflow into another repo?  

Sort of like calling CRAN to get R versions or getting package versions?

Sounds cool if that is the case.

**Comment by cicdguy** [comment]
@bms63 Yes exactly

**Comment by bms63** [comment]
Awesome.  `admiralci` sounds good to me, but should we make it more general like `pharmaci`.  I could see it being really handy for people wanting to use these workflows in non-admiral  packages but still pipeline-based packages.  

Be really cool outgrowth of this project that we have a recommended ci pipeline for R packages doing clinical reporting.

**Comment by cicdguy** [comment]
Nah I wouldn't call it pharmaci, simply because these are overly admiral-specific. Other pharma R projects I work on use completely different pipelines. I wouldn't put this in the one-size fits all category. Different teams prefer different approaches to development.

**Comment by bms63** [comment]
> @bms63 or rather than a new vignette why not add to https://pharmaverse.github.io/admiral/articles/pr_review_guidance.html#review-criteria?

I will make an issue to add this to the PR guidance

**Comment by bundfussr** [comment]
@dinakar29 , do these workflows also work in gitlab or do we need separate ones for gitlab?

**Comment by cicdguy** [comment]
> @dinakar29 , do these workflows also work in gitlab or do we need separate ones for gitlab?

Completely separate.

**Pull request merged**